store/multi_writer_app_store.go

```diff
@@ -97,6 +98,67 @@ func NewMultiWriterAppStore(appStore *IAVLStore, evmStore *EvmStore, saveEVMStat
 	return store, nil
 }
 
+func LoadMultiWriterAppStore(appStoreCfg *AppStoreConfig, evmstoreCfg *EvmStoreConfig, appStoreDB db.DBWrapper,
+	targetVersion int64, metricsDatabase bool, rootPath string) (*MultiWriterAppStore, error) {
+	iavlStore, err := NewIAVLStore(appStoreDB, appStoreCfg.MaxVersions, targetVersion, appStoreCfg.IAVLFlushInterval)
+	if err != nil {
+		return nil, err
+	}
+	appStoreEvmRoot := iavlStore.Get(rootKey)
+	evmDB, err := db.LoadDB(
+		evmstoreCfg.DBBackend,
+		evmstoreCfg.DBName,
+		rootPath,
+		evmstoreCfg.CacheSizeMegs,
+		evmstoreCfg.WriteBufferMegs,
+		metricsDatabase,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	evmStore, err := loadEvmStore(evmDB, iavlStore.Version(), evmstoreCfg.NumCachedRoots)
+	if err != nil {
+		return nil, err
+	}
+	evmStoreEvmRoot, evmVersion := evmStore.getLastSavedRoot(iavlStore.Version())
+	if !bytes.Equal(appStoreEvmRoot, evmStoreEvmRoot) {
+		log.Info(
+			"EVM root mismatch, resyncing roots...",
+			"evm.db-ver", evmVersion, "app.db-ver", iavlStore.Version(),
+			"evm.db-root", hex.EncodeToString(evmStoreEvmRoot), "app.db-evm-root", hex.EncodeToString(appStoreEvmRoot),
+		)
+		latestVersion, err := iavl.NewMutableTree(appStoreDB, 10000).LatestVersion(evmVersion)
+		if err != nil {
+			return nil, err
+		}
+		if latestVersion == 0 {
+			latestVersion = -1
+		}
+		iavlStore, err = NewIAVLStore(appStoreDB, appStoreCfg.MaxVersions, latestVersion, appStoreCfg.IAVLFlushInterval)
+		if err != nil {
+			return nil, err
+		}
+		evmStore.CloseDB()
+		evmDB, err = db.LoadDB(
+			evmstoreCfg.DBBackend,
+			evmstoreCfg.DBName,
+			rootPath,
+			evmstoreCfg.CacheSizeMegs,
+			evmstoreCfg.WriteBufferMegs,
+			metricsDatabase,
+		)
```

Comment:
Why do we need to close and reload the DB here?

Reply:
Previously, When I trying to reload `evmStore` with function `loadEvmStore` errors resource in used will occurs  so I need to close the `db` before reloading it again. 

```golang
func loadEvmStore(cfg *config.Config, targetVersion int64) (*store.EvmStore, error) {
	evmStoreCfg := cfg.EvmStore
	db, err := cdb.LoadDB(
		evmStoreCfg.DBBackend,
		evmStoreCfg.DBName,
		cfg.RootPath(),
		evmStoreCfg.CacheSizeMegs,
		evmStoreCfg.WriteBufferMegs,
		cfg.Metrics.Database,
	)
}
```

But right now when I move to `store/multi_writer_app_store.go` we only load `evmdb` once so it should be fine. I will remove it.